### PR TITLE
prometheus_metrics: exposed request timeout configuration

### PR DESCRIPTION
**Pull request by allanliu**
The prometheus_targets configuration now exposes a request timeout field (in seconds).  If there is no timeout explicitly set, then a default value of 1 second will be used.

### CODE REVIEW

**Comment by theopolis** [comment]
Why is `timeout` an `int` and not a `size_t`?

**Comment by allanliu** [comment]
There is no reason.  Will change.

**Comment by allanliu** [comment]
Updated to use size_t.

File: osquery/tables/applications/posix/prometheus_metrics.cpp [hunk 1]
```diff
@@ -116,7 +116,10 @@ QueryData genPrometheusMetrics(QueryContext& context) {
     sr[url.second.data()] = PrometheusResponseData{};
   }
 
-  scrapeTargets(sr);
+  int timeout = config.get_child("timeout", boost::property_tree::ptree())
```
Comment (theopolis) [inline osquery/tables/applications/posix/prometheus_metrics.cpp]: This should be a `size_t` too, right?

File: osquery/tables/applications/posix/prometheus_metrics.cpp [hunk 2]
```diff
@@ -116,7 +116,10 @@ QueryData genPrometheusMetrics(QueryContext& context) {
...
+                    .get_value<int>(1);
```
Comment (theopolis) [inline osquery/tables/applications/posix/prometheus_metrics.cpp]: As should this template type?

File: osquery/tables/applications/posix/prometheus_metrics.cpp (hunk 1 above)
Comment (allanliu) [inline osquery/tables/applications/posix/prometheus_metrics.cpp]: whops.  good catch. fixing

File: osquery/tables/applications/posix/prometheus_metrics.cpp (hunk 2 above)
Comment (allanliu) [inline osquery/tables/applications/posix/prometheus_metrics.cpp]: yup yup

**Comment by theopolis** [comment]
ok to test

**Pull request merged**